docs/design/interpreter/stackwalk.md

```diff
@@ -0,0 +1,25 @@
+# Stack walking for GC, exception handling and debugger
+To enable seamless integration of interpreter with GC, exception handling and debugger, we need to be able to walk stack that contains interleaved sequences of interpreted and AOT / JIT compiled frames. Fortunately, the same StackFrameIterator is used by the GC, exception handling and debugger, so adding a support to walk stack containing interpreter frames to this StackFrameIterator will give us support for all of the three scenarios.
+
+Here is the list of necessary changes:
+* Add a new explicit frame, the InterpreterFrame. Instance of this frame would be created in the mono_interp_exec_method. It will enable stack frame iterator to move through the interpreted frames managed by that mono_interp_exec_method. Please note that interpreter calls to other interpreted code are not recursively calling mono_interp_exec_method and each interpreted frame is represented by an instance of InterpFrame (these form a linked list).
```

Comment:
Are `InterpreterFrame` and `InterpFrame` different here? Not trying to quibble over names, but using names like this in these design documents is very difficult to parse. I find this sort of long form description difficult to understand - too many words and missing the logic.

As a suggestion in general, having basic steps (1), (2), ... (N) is much easier to follow and avoids encoding steps in long form descriptions.

Reply:
They are different. InterpFrame is a currently existing data structure in the Mono interpreter while the InterpreterFrame will be a new explicit frame derived from Frame in coreclr. I think we will end up renaming the InterpFrame to something else and use the InterpreterFrame for the explicit one, but I didn't want to rename the Mono structure for the sake of this document and I didn't have a better idea for naming of the new InterpreterFrame.